src/ReactCrop.tsx

```diff
@@ -596,6 +608,88 @@ class ReactCrop extends PureComponent<ReactCropProps, ReactCropState> {
     }
   };
 
+  onHandlerKeyDown = (
+    e: React.KeyboardEvent<HTMLDivElement>,
+    handle: 'nw' | 'n' | 'ne' | 'e' | 'se' | 's' | 'sw' | 'w'
+  ) => {
+    const { crop, disabled, minWidth = 0, maxWidth, minHeight = 0, maxHeight, onChange } = this.props;
+    const { width, height } = this.mediaDimensions;
+
+    // Keep the event from bubbling up to the container
+    if (e.key === 'ArrowUp' || e.key === 'ArrowDown' || e.key === 'ArrowLeft' || e.key === 'ArrowRight') {
+      e.stopPropagation();
+      e.preventDefault();
+    } else {
+      return;
+    }
+
+    // Cardinal directions only support 2 arrow keys each
+    if (
+      ((e.key === 'ArrowUp' || e.key === 'ArrowDown') && (handle === 'e' || handle === 'w')) ||
+      ((e.key === 'ArrowLeft' || e.key === 'ArrowRight') && (handle === 's' || handle === 'n'))
+    ) {
+      return;
+    }
+
+    if (disabled) {
+      return;
+    }
+
+    let changed = false;
+
+    if (!isCropValid(crop)) {
+      return;
+    }
+
+    const nextCrop = this.makeNewCrop();
+    const ctrlCmdPressed = navigator.platform.match('Mac') ? e.metaKey : e.ctrlKey;
+    const offset = ctrlCmdPressed
+      ? ReactCrop.nudgeStepLarge
+      : e.shiftKey
+      ? ReactCrop.nudgeStepMedium
+      : ReactCrop.nudgeStep;
+    let newWidth;
+    let newHeight;
+
+    // TODO: The crop area will leave the designated area (would be valid if moved vertically etc)
+    // TODO: Probably shouldn't expand farther to the right after already hitting the edge since that's the wrong direction
+
+    if (e.key === 'ArrowLeft' && (handle === 'nw' || handle === 'w' || handle === 'sw')) {
```

Comment:
I think in these conditions just change the width and height without worrying about clamping. Then after all adjustments have been made we can ensure the crop is in bounds (using existing method `containCrop`). Otherwise this function is going to get messy.

Reply:
I'm still learning how all of the functions work, but I think what you're saying makes sense.  Let me know if I still get it wrong.

Reply:
I think I am now using your advice.  I found that containCrop would give me invalidly large widths sometimes.  So I clamped the result and now it seems to work really well.

Reply:
Cool I can help with the containment stuff if the rest is in place. P.S. make sure you have latest master as I just changed `containCrop` to fix some edge cases

Reply:
It seems to work the way I'd expect it to now and I added the rest of the arrow handling in.  Let me know if you want anything else changed.

Reply:
I removed the clamp.  After merging master its working without the clamp.  Yay!

Reply:
Great!!